.github/workflows/nix_ci.yml

```diff
@@ -0,0 +1,29 @@
+name: Nix Buffers CI
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  check_flake_w_nix_on_ubuntu:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: DeterminateSystems/nix-installer-action@main
+      - uses: DeterminateSystems/magic-nix-cache-action@main
+      - uses: DeterminateSystems/flake-checker-action@main
+      - name: Run `nix flake check`
+        run: nix flake check
+
+  check_flake_w_nix_on_mac:
+    runs-on: macos-latest
```

Comment:
MacOS runners cost $0.08/minute and your last build took 15 minutes. Let's make sure this job runs quickly if it is to be used when we merge, both for iteration reasons and cost reasons (if each buffrs mr takes 5 attempts, it feels unreasonable for this to cost $12.

Reply:
When we're pulling from cache, we can finish the job in 1m. So it looks like the mandatory cost of a job rn is 0.08$, and any extra is incurred when there is more work to do.

Reply:
I've now set the mac CI run to only happen after a merge